virtcontainers/utils/utils_linux.go

```diff
@@ -87,3 +89,53 @@ func FindContextID() (*os.File, uint32, error) {
 	vsockFd.Close()
 	return nil, 0, fmt.Errorf("Could not get a unique context ID for the vsock")
 }
+
+// GetDevFormat get the formated filesystem of input disk use `blkid`, Such as `ext4`,`xfs`,etc.
+func GetDevFormat(disk string) (string, error) {
+	// refer to https://github.com/kubernetes/kubernetes/blob/v1.12.2/pkg/util/mount/mount_linux.go#L512
+	args := []string{"-p", "-s", "TYPE", "-s", "PTTYPE", "-o", "export", disk}
+	dataOut, err := exec.Command("blkid", args...).Output()
```

Comment:
Do we really need to rely on an external binary?

Reply:
**Iff** we have to run an external command, it may be better to use `lsblk --json` ?

Reply:
Can this be implemented in code instead , rather an execing into a binary.

Reply:
WDYT @zklei

Reply:
I haven't found a better way yet, sigh~@.@

Reply:
I had a quick trawl around and there doesn't seem to be a standard golang package to provide this. I did find:

- https://github.com/LDCS/qslinux/blob/master/blkid

... but that's under a GPL-3.0 license.

The standard approach does seem to be to run `blkid(8)` and parse the output:

- https://github.com/kubernetes/kubernetes/blob/master/pkg/util/mount/mount_linux.go#L512
- https://github.com/google/embiggen-disk/blob/master/part.go#L87